src/app/modules/item/services/item-task-answer.service.ts

```diff
@@ -145,4 +139,26 @@ export class ItemTaskAnswerService implements OnDestroy {
   clearAnswer(): Observable<unknown> {
     return this.task$.pipe(take(1), switchMap(task => task.reloadAnswer('')));
   }
+
+  private saveAnswerAndStateInterval(initialAnswer: string, initialState: string): Observable<{ success: boolean }> {
+    const saved = { answer: initialAnswer, state: initialState };
+    return interval(answerAndStateSaveInterval).pipe(
+      switchMapTo(this.task$),
+      switchMap(task => forkJoin({ answer: task.getAnswer(), state: task.getState() })),
+      distinctUntilChanged((a, b) => a.answer === b.answer && a.state === b.state),
+      filter(({ answer, state }) => answer !== saved.answer || state !== saved.state),
+      take(1),
+      withLatestFrom(this.config$),
+      switchMap(([{ answer, state }, { route, attemptId }]) =>
+        this.currentAnswerService.update(route.id, attemptId, { answer, state }).pipe(
+          map(() => {
+            Object.assign(saved, { answer, state });
```

Comment:
this is a side effect, it should be done in a ` tap` so that it is clean that it is a dirty side effect